hadoop-hdfs-project/hadoop-hdfs-native-client/src/main/native/libhdfs/hdfs.c

```diff
@@ -2498,6 +2498,11 @@ int hadoopRzOptionsSetByteBufferPool(
         return -1;
     }
 
+    if (opts->byteBufferPool) {
+        // Delete any previous ByteBufferPool we had.
+        (*env)->DeleteGlobalRef(env, opts->byteBufferPool);
```

Comment:
This should probably be moved down until right before setting the _new_ value. Otherwise, if we failed somewhere below, we'd end up with two issues: (1) the function would return an error but had a side effect of clearing the previous value, which seems unintuitive. (2) the function would still leave opts->byteBufferPool set, but with a now-invalid reference.

Reply:
Done. Fixed a few other issues I found with this method while I was at it.